lib/downloader.rb

```diff
@@ -136,16 +124,34 @@ def self.get_novel_data_dir_by_target(target)
   # target のIDを取得
   #
   def self.get_id_by_target(target)
-    toc_url = get_toc_url(target) or return nil
-    @@database.get_id("toc_url", toc_url)
+    data = get_data_by_target(target)
+    data && data["id"]
   end
 
   #
   # target からデータベースのデータを取得
   #
   def self.get_data_by_target(target)
-    toc_url = get_toc_url(target) or return nil
-    @@database.get_data("toc_url", toc_url)
+    target = Narou.alias_to_id(target)
+    case get_target_type(target)
+    when :url
+      setting = @@settings.find { |s| s.multi_match_once(target, "url") }
+      if setting
+        toc_url = setting["toc_url"]
+        return @@database.get_data_by_toc_url(toc_url, setting)
+      end
+    when :ncode
+      @@database.each do |_, data|
```

Comment:
__[RuboCop]__

Performance/HashEachMethods: Use `each_value` instead of `each`.


<!-- Comment by [SideCI](https://sideci.com) -->